src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package com.internet.shop.security;
+
+import com.internet.shop.exceptions.AuthenticationException;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.models.User;
+import com.internet.shop.services.UserService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User userFromDb = userService.findByLogin(login)
+                .orElseThrow(() -> new AuthenticationException("Incorrect user name or password"));
+        if (userFromDb.getPassword().equals(password)) {
```

Comment:
let's check all conditions inside `if` so that we could throw our Exception only once

Reply:
fixed